source/common/http/codec_client.h

```diff
@@ -180,8 +182,10 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
 
   // Network::ConnectionCallbacks
   void onEvent(uint32_t events) override;
-  void onAboveWriteBufferHighWatermark() override {}
-  void onBelowWriteBufferLowWatermark() override {}
+  // Pass watermark events from the connection on to the codec which will pass it to the underlying
+  // streams.
+  void onAboveWriteBufferHighWatermark() override { codec_->onAboveWriteBufferHighWatermark(); }
```

Comment:
I haven't gone over the full PR yet, but one question I have is how do the connection and stream watermark event edge triggers interact? For example, if connections goes above high watermark, then stream goes above high watermark, are we guaranteed to only ever get one `onAboveWriteBufferHighWatermark` for the stream's registered callbacks, or will we see two? I recall this was an issue I had to write some funky logic to handle when experimenting with this a while back.